Topics/Tech_Stacks/Unity_UI.md

```diff
@@ -0,0 +1,107 @@
+
+You can change the order of Game Object by simply drawing them in the Hierarchy window.
+
+## Customize your UI element!
+
+At this point, you should know how to create different kinds of UI element on the Canvas. Now it is the time to design them with your unique style!
+
+Here are some basics tools that you might need during the design.
+
+### The Rect Tool
+On the top left of the toolbar, you will find some rect tools that can help you manupulate your UI element.
+
+![GUI_Rect_Tool_Button](https://github.com/arielchen07/learning-software-engineering.github.io/assets/105243552/c0f7e468-d19f-43ee-9ba4-5f31230f9fb4)
+
+With that said, you can move, resize and rotate UI elements once you select the UI element. You can move the UI element by clicking somewhere inside the element and dragging it tp the place you want. You can resize it by clicking on the edges or corners and dragging. The element can be rotated by hovering the cursor slightly away from the corners until the mouse cursor looks like a rotation symbol. You can then click and drag in either direction to rotate.
+
+## Inspector
+
+If you double clicked the UI element in the Hierarchy window, you should have the Inspector window shown on your right. All the properties and components of the UI element should be editable in the Inspector window.
+
+<img width="343" alt="Screenshot 2023-11-23 at 6 42 01 PM" src="https://github.com/arielchen07/learning-software-engineering.github.io/assets/105243552/f64fe1d8-c602-4d20-ba19-f6a5085ffe8d">
+
+Most of the components are intuitive to understand and easy to modify, so here we will introduce one of the most important components for all Game Objects:
+
+### Rect Transform
+Rect Transform helps modify the position, rotation, and scale of the Game Object, while it also has a width and height attribute that can decide the dimensions of the UI element.
+
+**Pivot** and **Anchor** are two useful concept to understand here. 
+
+**Pivot**: Pivot is a small circle on the UI element. All the modifications like rotation, size, and scale, will based on the position of the pivot. You can try different positions of pivot to see how it affects the outcome of a rotation, resizing, or scaling. 
+
+**Anchor**: Anchors are shown as four small triangular in the Scene View. The anchoring allows the child to stretch together with the width or height of the parent. Each corner of the rectangle has a fixed offset to its corresponding anchor. This way the different corners of the rectangle can be anchored to different points in the parent rectangle. You can also think of Anchor as setting relative position to the parent object.
+
+![UI_Anchored1](https://github.com/arielchen07/learning-software-engineering.github.io/assets/105243552/f29a495b-a757-4417-bc4e-b06cb6872cf1)
+
+![UI_Anchored2](https://github.com/arielchen07/learning-software-engineering.github.io/assets/105243552/b0562aeb-a95a-4347-80f6-41d45ef03b27)
```

Comment:
For the pivot and anchor gifs, I think it would be more clear to label them/add picture captions to show which gif corresponds to pivot and which corresponds to anchor. Or you can place each gif directly after its own description instead of putting 2 together at the end.

Reply:
So this two gifs all contains both pivot and anchor, it just showing different cases. I also described them in the text. 
But your suggestions make sense. I've added captions in these two gifs. It is a bit hard to let the caption moving in the gif and text is lossy in the gif. Let me know if you think it is ok.

Reply:
ohh I see what you mean now. I think I misunderstood what pivot and anchor are the last time I read your document. I think it's a bit hard for me to find pivot and anchor in the gif from the word description. And I see that adding caption in the moving gif is a bit hard and could cause some confusion. 
So I think maybe you can add another screenshot (fixed image, not gif) and label pivot and anchor there? like this:
<img width="400" alt="image" src="https://github.com/learning-software-engineering/learning-software-engineering.github.io/assets/51133017/d28a65b3-4864-4201-83f0-097dfe88a5e3">
I'm not sure whether I labeled correctly, but I think you get the idea

Reply:
Well I think it is a bit redundant to add another fixed image showing this. The pictures will occupied a lot of space in the page.